### PR TITLE
remove DataSourceCredentials

### PR DESCRIPTION
**Pull request by xiangyan99**

### CODE REVIEW

**Comment by xiangyan99** [comment]
#10860

**Comment by xiangyan99** [comment]
Remove DataSourceCredentials
Just uplift ConnectionString to SearchIndexerDataSource

**Comment by xiangyan99** [comment]
rename SearchIndexerDataSource to SearchIndexerDataSourceConnection
See in #11565

File: sdk/search/azure-search-documents/azure/search/documents/indexes/_internal/_models.py [hunk 1]
```diff
@@ -193,3 +193,61 @@ def __init__(
...
+        'connection_string': {'required': True},
+        'container': {'required': True},
+    }
+
+    _attribute_map = {
+        'name': {'key': 'name', 'type': 'str'},
+        'description': {'key': 'description', 'type': 'str'},
+        'type': {'key': 'type', 'type': 'str'},
+        'connection_string': {'key': 'connectionString', 'type': 'str'},
+        'container': {'key': 'container', 'type': 'SearchIndexerDataContainer'},
```
Comment (rakshith91) [inline sdk/search/azure-search-documents/azure/search/documents/indexes/_internal/_models.py:235]: should this be "flattenned" too like the connection string? perhaps just take in a container name + kwargs and create the DataContainer ourselves?

File: sdk/search/azure-search-documents/azure/search/documents/indexes/_internal/_search_indexer_client.py [hunk 2]
```diff
@@ -269,21 +274,22 @@ def create_datasource(self, data_source, **kwargs):
...
-        result = self._client.data_sources.create(data_source, **kwargs)
-        return result
+        packed_data_source = pack_search_indexer_data_source(data_source)
+        result = self._client.data_sources.create(packed_data_source, **kwargs)
+        return unpack_search_indexer_data_source(result)
 
     @distributed_trace
     def create_or_update_datasource(self, data_source, name=None, **kwargs):
-        # type: (SearchIndexerDataSource, Optional[str], **Any) -> Dict[str, Any]
+        # type: (SearchIndexerDataSourceConnection, Optional[str], **Any) -> SearchIndexerDataSourceConnection
```
Comment (rakshith91) [inline sdk/search/azure-search-documents/azure/search/documents/indexes/_internal/_search_indexer_client.py:283]: We should do this in a different PR prolly - but we must allow individual keywords only to update the datasource - which if present would override the datasource. (See skillsets for example)
